admin/class-primary-term-admin.php

```diff
@@ -38,6 +39,51 @@ protected function get_current_id() {
 		return $post_id;
 	}
 
+	/**
+	 * Filter to add hidden fields for primary taxonomies to the metabox.
+	 *
+	 * @param {string} $content The metabox content.
+	 *
+	 * @return string The HTML content.
+	 */
+	public function add_input_fields( $content ) {
+		$taxonomies = $this->get_primary_term_taxonomies();
+
+		foreach ( $taxonomies as $taxonomy ) {
+			$primary_term = $this->get_primary_term( $taxonomy->name );
+			$content .= $this->primary_term_field( $taxonomy->name, $primary_term );
+			$content .= wp_nonce_field( 'save-primary-term',WPSEO_Meta::$form_prefix . 'primary_' . $taxonomy->name . '_nonce', false, false );
+		}
+		return $content;
+	}
+
+	/**
+	 * Generates the HTML for a hidden field for a primary taxonomy.
+	 *
+	 * @param {string} $taxonomy_name The taxonomy's slug.
+	 * @param {string} $primary_term  The primary term's id.
+	 *
+	 * @return string The HTML for a hidden primary taxonomy field.
+	 */
+	protected function primary_term_field( $taxonomy_name, $primary_term ) {
+		$output = '<input type="hidden" class="yoast-wpseo-primary-term"';
```

Comment:
You should use sprintf for this:
```
return sprintf( 
    '<input type="hidden" id="%1$s" name="%2$s" value="%3$s" )',
    esc_attr( $this->generate_field_id( $taxonomy_name ) ),
    esc_attr( WPSEO_Meta::$form_prefix ) . 'primary_' . $taxonomy_name . '_term"',
    esc_attr( $primary_term ) 
);    
```

I also want a method that generates the name. The same approach like you did for the field. Name it `generate_field_name`